migrations/20190212004845-tags-rlinks-not-null-multiple-models.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+const db = require('../db/index');
+const constants = require('../db/constants');
+
+const modelsAndCols = {
+  Aspect: ['tags', 'relatedLinks'],
+  Generator: ['tags'],
+  GeneratorTemplate: ['tags'],
+  Perspective:
+    ['aspectFilter', 'aspectTagFilter', 'subjectTagFilter', 'statusFilter'],
+  Sample: ['relatedLinks'],
+};
+
+function updateRowsAndColumns(qi, seq, modelName, colNamesArr) {
+  const whereArr = [];
+  colNamesArr.forEach((cname) => {
+    const o = {};
+    o[cname] = { $eq: null };
+    whereArr.push(o);
+  });
+
+  const model = db[modelName];
+  return model.findAll({ // find records with any of given columns = null
+    where: { $or: whereArr },
+  })
+    .then((records) => Promise.all(
+      records.map((record) => {
+        const toUpdate = {};
+        colNamesArr.forEach((colName) => {
+          if (record[colName] === null) {
+            toUpdate[colName] = []; // set the record column to empty array
+          }
+        });
+
+        return record.update(toUpdate);
+      })
+    ))
+    .then(() => Promise.all(colNamesArr.map((colName) => {
+      if (colName === 'relatedLinks') {
+        return qi.changeColumn(`${modelName}s`, colName, {
+          type: seq.ARRAY(seq.JSON),
+          allowNull: false, // change relatedLinks column to allowNull: false
+          defaultValue: constants.defaultJsonArrayValue,
```

Comment:
Is it necessary to include the type and defaultValue even though they aren"t changing?

Reply:
For changeColumn, the sequelize docs says:

> This method changes the meta data of an attribute. It is possible to change the default value, allowance of null or the data type. Please make sure, that you are completely describing the new data type.